src/effects/native/phasereffect.cpp

```diff
@@ -0,0 +1,170 @@
+                           const EffectManifest& manifest) 
+        : m_pStagesParameter(pEffect->getParameterById("stages")),
+          m_pFrequencyParameter(pEffect->getParameterById("frequency")),
+          m_pDepthParameter(pEffect->getParameterById("depth")),
+          m_pFeedback(pEffect->getParameterById("feedback")),
+          m_pSweepWidth(pEffect->getParameterById("sweep_width")) {
+    Q_UNUSED(manifest);
+}
+
+PhaserEffect::~PhaserEffect() {
+    //qDebug() << debugString() << "destroyed";
+}
+
+void PhaserEffect::processChannel(const ChannelHandle& handle,
+                                  PhaserGroupState* pState,
+                                  const CSAMPLE* pInput, CSAMPLE* pOutput,
+                                  const unsigned int numSamples,
+                                  const unsigned int sampleRate,
+                                  const EffectProcessor::EnableState enableState,
+                                  const GroupFeatureState& groupFeatures) {
+
+    Q_UNUSED(handle);
+    Q_UNUSED(enableState);
+    Q_UNUSED(groupFeatures);
+    Q_UNUSED(sampleRate);
+
+    CSAMPLE frequency = m_pFrequencyParameter->value();
+    CSAMPLE depth = m_pDepthParameter->value();
+    CSAMPLE feedback = m_pFeedback->value();
+    CSAMPLE sweepWidth = m_pSweepWidth->value();
+    int stages = m_pStagesParameter->value();
+
+    CSAMPLE* filterLeft = pState->filterCoefLeft;
+    CSAMPLE* filterRight = pState->filterCoefRight;
+    CSAMPLE* oldInLeft = pState->oldInLeft;
+    CSAMPLE* oldOutLeft = pState->oldOutLeft;
+    CSAMPLE* oldInRight = pState->oldInRight;
+    CSAMPLE* oldOutRight = pState->oldOutRight;
+
+    CSAMPLE left = 0, right = 0;
+    CSAMPLE leftPhase; 
+    CSAMPLE rightPhase;
+    CSAMPLE freqSkip = frequency * 2.0 * M_PI / sampleRate;
+
+    int counter = 0;
+    int updateCoef = 32;
+
+    const int kChannels = 2;
+    for (unsigned int i = 0; i < numSamples; i += kChannels) {
+
+        pState->time++;
+        left = pInput[i] + left * feedback; 
+        right = pInput[i + 1] + right * feedback;
+
+        leftPhase = fmodf(freqSkip * pState->time, 2.0 * M_PI);
+        rightPhase = fmodf(freqSkip * pState->time + M_PI, 2.0 * M_PI);
+
+        if ((counter++) % updateCoef == 0) {
+            for (int j = 0; j < stages; j++) {
+                CSAMPLE delayLeft = 0.5 + 0.5 * sin(leftPhase);
+                CSAMPLE delayRight = 0.5 + 0.5 * cos(rightPhase);
+                
+                delayLeft = min((double)(sweepWidth * delayLeft), 0.99 * M_PI);
+                delayRight =  min((double)(sweepWidth * delayRight), 0.99 * M_PI);
+
+                delayLeft = tan(delayLeft / 2);
+                delayRight = tan(delayRight / 2);
+
+                filterLeft[j] = (1.0 - delayLeft) / (1.0 + delayLeft);
+                filterRight[j] = (1.0 - delayRight) / (1.0 + delayRight);
+            }
+        }
+
+        for (int j = 0; j < stages; j++) {
```

Comment:
This is the iir filter loop.
I would prefer to reformat it similar to this: 
https://github.com/mixxxdj/mixxx/blob/master/src/engine/enginefilteriir.h#L268 
Or this 
https://github.com/mixxxdj/mixxx/blob/master/src/engine/enginefiltermoogladder4.h#L151
To have a reference to the used iir model. 
This will help to recognize the form DF1 / DF2